src/main/java/seedu/apollo/module/Timetable.java

```diff
@@ -61,6 +61,65 @@ public ArrayList<Integer> getWeeks() {
         return weeks;
     }
 
+    /**
+     * This method compresses the weeks of a lesson into a String that can be used for the UI.
+     * For example, if a lesson occurs on weeks 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, then
+     * the method will return [Weekly]. If a lesson does not occur for consecutive weeks, then
+     * it will print out all weeks that the lesson occurs on. For example, if a lesson occurs on
+     * weeks 5, 7, 9, 11, then the method will return [Weeks: 5, 7, 9, 11, 13]. Lastly if a lesson occurs on
+     * consecutive weeks but does not span the entire semester, then the method will return the first and last weeks.
+     * For example, if a lesson occurs on weeks 1, 2, 3, 4, 5, then the method will return [Weeks: 1-5].
+     * @param   timetable       The timetable of the lesson.
+     * @return  The formatted String representation of the weeks.
+     */
+    public String compressedWeeks(Timetable timetable) {
+        ArrayList<Integer> unfilteredWeeks = timetable.getWeeks();
+        ArrayList<Integer> filteredWeeks = new ArrayList<>();
+        String formattedWeeks;
+
+        int firstWeek = unfilteredWeeks.get(0);
+        int lastWeek = unfilteredWeeks.get(unfilteredWeeks.size() - 1);
+
+        //First, check if the lesson spans the whole 13-week semester
+        if (unfilteredWeeks.size() == 13) {
+            formattedWeeks = "[Weekly]";
+            return formattedWeeks;
```

Comment:
oh actl if u want to nitpick for efficiency, u can check this immediately after line 76 and `return "[Weekly]"` if true to skip doing all the firstWeek lastWeek stuff